packages/indicators/src/builders/buildArithmetic/aggregationConfig.ts

```diff
@@ -0,0 +1,145 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { getVariables } from '@beyondessential/arithmetic';
+
+import { toArray } from '@tupaia/utils';
+import { ArithmeticConfig, AggregationDescriptor, AggregationSpecs } from './types';
+
+enum AggregationType {
+  String, // 'SUM'
+  Array, // ['SUM', 'MOST_RECENT']
+  Object, // { type: 'OFFSET_PERIOD', config: { periodType: 'week' }}
+  // Dictionary keys are data elements included in `ArithmeticConfig.formula`
+  // or `ArithmeticConfig.parameters`
+  Dictionary, // { BCD1: 'SUM', BCD2: ['COUNT', 'FINAL_EACH_WEEK' ] }
+  Invalid,
+}
+
+const isParameterCode = (parameters: { code: string }[], code: string) =>
+  !!parameters.find(p => p.code === code);
+
+const getAggregationType = (aggregation: unknown): AggregationType => {
+  switch (typeof aggregation) {
+    case 'string':
+      return AggregationType.String;
+    case 'object':
+      if (aggregation === null) {
+        return AggregationType.Invalid;
+      }
+      if (Array.isArray(aggregation)) {
+        return AggregationType.Array;
+      }
+      if ('type' in aggregation) {
+        return AggregationType.Object;
+      }
+      return AggregationType.Dictionary;
+    default:
+      return AggregationType.Invalid;
```

Comment:
If we already know the aggregation type is invalid at this point, perhaps it's best just to throw an error here instead of having this type on the enum?

Reply:
Good suggestion! Will do that